fe-public/src/components/Calendar.astro

```diff
@@ -0,0 +1,37 @@
+---
+import ical from 'node-ical';
+let URL = import.meta.env.ICAL || "https://ieconnects.ie.edu/ical/ical_club_300003041.ics"
+const webEvents = await ical.async.fromURL(URL);
+let events = Object.values(webEvents).map((event) => {
+  return {
+    title: event.summary?.val,
+    start: new Date(event.start),
+    url: event.url,
+    description: event.description
+  };
+})
+events = events.sort((a, b) => {
+  return a.start - b.start;
+}).filter((event) => {
+  return event.start > new Date();
```

Comment:
Rather than mercilessly removing all past events we could show them in some other format.